frontend/src/utils/generate-warnings.ts

```diff
@@ -241,11 +242,13 @@ export function produceRequirementGroupWarning(
     }
   }
 
-  let requirementGroups: string[] = major.requirementGroups;
   let res: IRequirementGroupWarning[] = [];
-  for (const name of requirementGroups) {
-    let requirementGroup: IMajorRequirementGroup =
-      major.requirementGroupMap[name];
+
+  const sortedRequirements = sortOnValues(
+    Object.values(major.requirementGroupMap)
+  );
+
+  for (const requirementGroup of sortedRequirements) {
     // todo: if req group is RangeSection or contains a ICourseRange process after all other requirement groups.
     // this is because these reqs can be satisfied by a wide range of courses, and you don't want it using up
     // a course that is need to satisfy a more constrained requirement.
```

Comment:
I believe the sorting within this PR is what this comment is referring to, if so we can remove it